modules/common/src/main/Camo.scala

```diff
@@ -0,0 +1,14 @@
+package lila.common
+
+final class Camo(endpoint: String, secret: String) {
+  def apply(url: String): String = {
+    if (secret == "") url
+    else {
+      val mac = javax.crypto.Mac.getInstance("HMACSHA1")
+      mac.init(new javax.crypto.spec.SecretKeySpec(secret.getBytes, "HMACSHA1"))
```

Comment:
do we need to instanciate and initialize a mac instance every time?

Reply:
seems easiest, because it has mutable state.